### ISSUE TITLE
[Support] Slicing Failed to showing record for next page

### ISSUE THREAD

**Issue by hans0801** [DirectoryTree/LdapRecord#619]
**Environment:**
 - LDAP Server Type: ActiveDirectory
 - PHP Version: 7.4

Hi @stevebauman ,

Firstly I will say thank you so much for your great efforts for build this great lib.
Very helpfull to me for manage my AD Account from php web base.

I tried to use Slicing for get list of all users from my Active Directory server.
I can get the data, but when 1 clicked on next page then the record on the list isn't update. Still showing up the list from Page 1.
Here is my code:

`#Active Directory LDAP Connection
        $this->model->createADConnection();

        $connection = Container::getDefaultConnection();

        $query = $connection->query()->where([
          ['objectclass', '=', 'top'],
          ['objectclass', '=', 'person'],
          ['objectclass', '=', 'organizationalperson'],
          ['objectclass', '=', 'user'],
        ]);

        $users = $query->slice($page = 1, $perPage = 10);

        $data = array();

        foreach ($users as $user) {
            $row = array();
            $row[] = '<span class="text-primary">'.ucfirst(strtolower($user['samaccountname'][0]).'</span>');
            $row[] = $user['cn'][0];

            $str_user = User::find($user['distinguishedname'][0]);
            $uac = new AccountControl(
              $str_user->getFirstAttribute('userAccountControl')
            );
            if($uac=="512" || $uac=="66048") # 512=Normal Account, 512+2(514)=Account Disabled, 8388608=Password Expired
            {
              $row[] = '<p style="color:green;">Normal</p>';
            }elseif ($uac=="514" || $uac=="66050")
            {
              $row[] = '<p style="color:red;">Disabled</p>';
            }elseif ($uac=="8389120")
            {
              $row[] = '<p style="color:yellow;">Password Expired</p>';
            }
            $row[] = '
                      <div class="btn-group" role="group" aria-label="Basic example">
                          <a href="'.url("ad_management/view_user/".enc_url($user['samaccountname'][0])).'" id="view" class="btn btn-primary" title="'.cclang("detail").'">
                            <i class="ti-file"></i>
                          </a>
                          <a href="'.url("ad_management/update/".enc_url($user['samaccountname'][0])).'" id="update" class="btn btn-warning" title="'.cclang("update").'">
                            <i class="ti-pencil"></i>
                          </a>
                          <a href="'.url("ad_management/delete/".enc_url($user['samaccountname'][0])).'" id="delete" class="btn btn-danger" title="'.cclang("delete").'">
                            <i class="ti-trash"></i>
                          </a>
                        </div>
                     ';
  
            $data[] = $row;
        }
  
        $output = array(
                        "draw" => $_POST['draw'],
                        "recordsTotal" => $users->total(),
                        "recordsFiltered" => $users->total(),
                        "data" => $data,
                );
        //output to json format
        return $this->response($output);`

Please kindly your help.

Thanks in advance

Best,
hans0801

**Comment by stevebauman** [comment]
Hi @hans0801,

How are you incrementing the page integer when the user navigates to the next page? Is the below code accurate from your example?

```php
$users = $query->slice($page = 1, $perPage = 10);
```

**Comment by hans0801** [comment]
Hi @stevebauman ,

 Apologies before that if my code was wrong.
I'm used this code for trying to navigates the next page. 
`$output = array(
                    "draw" => $_POST['draw'],
                    "recordsTotal" => $users->total(),
                    "recordsFiltered" => $users->total(),
                    "data" => $data,
            );`
Thanks

**Comment by stevebauman** [comment]
Hey @hans0801, that unfortunately doesn't answer my question above.

In your code example, you've hard-coded the `$page` to `1`. Typically the `$page` integer should be supplied by a query or post parameter. If you don't ask for page 2, you will always see page 1. Ex:

```php
// Before:
$users = $query->slice($page = 1, $perPage = 10);

// After:
$users = $query->slice($_POST['page'], $perPage = 10);
```

**Comment by hans0801** [comment]
Hi @stevebauman ,
actually for now i'm used this code for the slicing:
`
        #Active Directory LDAP Connection
        $this->model->createADConnection();

        $connection = Container::getDefaultConnection();

        $query = $connection->query()->where([
          ['objectclass', '=', 'top'],
          ['objectclass', '=', 'person'],
          ['objectclass', '=', 'organizationalperson'],
          ['objectclass', '=', 'user'],
        ]);

        $users = $query->slice(**$this->input->post('valCurrentPage')?? 1**, $perPage = 10);`

and here is my code from views for get the 'valCurrentPage' :
`
table = $('#table').DataTable({
      "drawCallback": function(){
      $('.paginate_button', this.api().table().container())          
         .on('click', function(){
          var info = table.page.info();
          **$('#valCurrentPage').val(info.page+1);**
          $table.ajax.reload();
         });       
      }
`
but i still failed to get the record for next page.
any advise?

Thanks in advance

**Comment by hans0801** [comment]
Will finding another method. 
Thanks
